### ISSUE TITLE
Add Python host programs to more samples.

### ISSUE THREAD

**Issue by cgranade** [microsoft/Quantum#233]
Currently, most samples only have a C# host program. It would be nice if more or even most samples demonstrated how to use Python to call into Q#.

**Comment by bromeg** [comment]
We are working on a samples re-organization, it would be best for those interesting in helping with this issue to work on it after #235 has been merged.

**Comment by cgranade** [comment]
As per @bromeg's comment above, I think we're in a better place to add host programs now that #235 is in. ♥

**Comment by deadmau6** [comment]
Hello, are you still looking for help with this issue? Also this is my first time contributing to Open Source but this seems like a simple issue.

**Comment by cgranade** [comment]
@deadmau6: Thanks for your patience, we'd definitely appreciate help with this one! We recently added several Python host programs to the samples in the [`getting-started` folder](../blob/master/samples/getting-started), but there's a number of samples that still only have C# host programs. A good place to start might be with the [`bit-flip-code` sample](https://github.com/microsoft/Quantum/blob/master/samples/error-correction/bit-flip-code/), as adding a Python host to that would require fairly minimal changes. Please let me know if there's anything I can do to help, in the meantime we really appreciate the contribution! 💕

**Comment by deadmau6** [comment]
**Update:**
Just giving a quick update here, so I finished the `bit-flip-code` and almost all of the samples in `algorithms/`. I did have an issue with `order-finding` in which the solution is maybe not the best and I can go into more detail either here or in a PR. Anyways I should be able to get a lot more done over the weekend because I don't have to work.

**Questions:**
I do have a quick question, should I submit all of the samples in one PR? Also is there some sort of `issues` and `sub-issues` list in github?

I'm just thinking it would be easier to provide updates if there was a task list I could check off like `algorithms -> done`. Thanks!

**Comment by cgranade** [comment]
Thanks, @deadmau6! For `order-finding`, if it would be helpful I'm happy to take a look in a PR and offer what suggestions I can. As for PR style, I don't think there's a huge advantage either way for having one big PR or several small ones in this case; whatever works best for you.

Thanks for the suggestion of a sub-issues list, probably the easiest way to do that is for me to edit the original issue with `- [ ]` lists, as GitHub then renders that as a kind of sub-issue. I'll go on and do that now, then. Thanks!

**Comment by deadmau6** [comment]
@cgranade, quick update, sorry it has taken me so long. I am nearly complete with all of the samples, but I hit a wall with most of the chemistry samples. I feel like I am missing something like a library, but I have everything done except for `GetGateCount`, `CreateHubbardHamiltonian`, `SimulateHubbardHamiltonian`, `RunSimultation`, and `MolecularHydrogen`. Anyways can I submit a PR for what I have and then could you maybe help me out with the chemistry stuff?

**Comment by cgranade** [comment]
My apologies for the delay, @deadmau6, just getting spun back up after a vacation. In any case, the chemistry host programs are definitely the hardest to adapt, as there's a lot of pre- and post-processing that happens in the classical hosts to load problem descriptions. Especially given planned simplifications such as #203, I agree with making a PR for the other host programs for now, that sounds like a great idea. Thank you again for all your help!

**Comment by deadmau6** [comment]
Awesome! And no worries I'll submit the PR once I get home tonight.

**Comment by cgranade** [comment]
Thanks again for your contribution, @deadmau6! I think that brings us down to just missing Python hosts for the chemistry samples, so I'll go on and update the title/description accordingly.

**Comment by sdf94** [comment]
@cgranade Can you assign me to CreateHubbardHamiltonian?

**Comment by superkirill** [comment]
@cgranade, is SimulateHubbardHamiltonian still open for contributions?

**Comment by ShaunJW360** [comment]
@cgranade Would it be possible to contribute?

**Comment by cgranade** [comment]
> @cgranade, is SimulateHubbardHamiltonian still open for contributions?

Yep! If you're interested in contributing, the relevant folder is at https://github.com/microsoft/Quantum/tree/master/samples/chemistry/SimulateHubbardHamiltonian. That sample currently only has a [C# host](https://github.com/microsoft/Quantum/blob/master/samples/chemistry/SimulateHubbardHamiltonian/Program.cs), but it should be adaptable to a Python host since it doesn't rely on project references.
Thanks for the help, @superkirill!

> @cgranade Would it be possible to contribute?

Absolutely,  thank you for the help @ShaunJW360! Please let me know if there's anything I can do to help with contributions; in the meantime, the [contribution guide](https://docs.microsoft.com/quantum/resources/contributing/) may be of interest. Thanks!

**Comment by superkirill** [comment]
Yes, I want to take a look at this then. Thanks, @cgranade!
